File: frontend/src/types/domain.ts
export type Instance = {
  id?: string
  instance_name: string
  phone_number?: string | null
  profile_name?: string | null
  status?: string | null
  provider?: string
  synced_at?: string | null
}

export type Contact = {
  id: string
  remote_jid: string
  push_name?: string | null
  phone_number?: string | null
  is_business: boolean
}

export type Chat = {
  id: string
  remote_jid: string
  chat_name?: string | null
  is_group: boolean
  unread_count: number
  last_message_at?: string | null
}

export type Message = {
  id: string
  message_external_id: string
  remote_jid: string
  from_me: boolean
  sender_name?: string | null
  content?: string | null
  message_type: string
  timestamp: string
}

export type SemanticResult = {
  message_id: string
  remote_jid: string
  content?: string | null
  timestamp: string
  score: number
}

File: backend/app/schemas/domain.py
from datetime import datetime

from pydantic import BaseModel


class InstanceOut(BaseModel):
    id: str
    instance_name: str
    phone_number: str | None = None
    profile_name: str | None = None
    status: str | None = None
    provider: str
    synced_at: datetime | None = None


class ContactOut(BaseModel):
    id: str
    remote_jid: str
    push_name: str | None = None
    phone_number: str | None = None
    is_business: bool = False


class ChatOut(BaseModel):
    id: str
    remote_jid: str
    chat_name: str | None = None
    is_group: bool = False
    unread_count: int = 0
    last_message_at: datetime | None = None


class MessageOut(BaseModel):
    id: str
    message_external_id: str
    remote_jid: str
    from_me: bool
    sender_name: str | None = None
    content: str | None = None
    message_type: str
    timestamp: datetime


class SyncResponse(BaseModel):
    instance_name: str
    contacts_count: int
    chats_count: int
    messages_count: int
    embedded_count: int
